challenges/functions.js

```diff
@@ -7,24 +7,41 @@
   * In the body of the function return the callback with the two parameters that you created
 */
 
+ function consume(item1, item2 , callback) {
+   return callback(item1, item2);
+ }
 
 /* Step 2: Create several functions to callback with consume();
   * Create a function named add that returns the sum of two numbers
   * Create a function named multiply that returns the product of two numbers 
   * Create a function named greeting that accepts a first and last name and returns "Hello first-name last-name, nice to meet you!"
 */
+function add(x, y , cb) {
+  return cb = x + y;
+}
+console.log(consume(1, 3, add));
+
+function multiply(x, y, cb) {
+  return cb = x * y;
+}
+console.log(consume(4, 5, multiply));
+
+const greeting = (firstName, lastName, cb) => { 
+  return `${"Hello"} ${firstName} ${lastName} ${"nice to meet you!"}`
+}
+console.log(consume('Javier', 'Alvarez', greeting));
 
 
 /* Step 3: Check your work by uncommenting the following calls to consume(): */
-//consume(2,2,add); // 4
-// consume(10,16,multiply); // 160
-// consume("Mary","Poppins", greeting); // Hello Mary Poppins, nice to meet you!
+console.log(consume(2,2,add)); // 4
+console.log(consume(10,16,multiply)); // 160
+console.log(consume("Mary","Poppins", greeting)); // Hello Mary Poppins, nice to meet you!
 
 
 // ==== Closures ==== 
 // Explain in your own words why `nestedfunction()` can access the variable `internal`.
 
-// Explanation: 
+// Explanation:  nestedfunction has access to because child fucntions have access to thier parent function scope, additionally parents do not have access to their nested children scope.
 
```

Comment:
Show me some better examples, I want to be able to show my kids and have them understand it.